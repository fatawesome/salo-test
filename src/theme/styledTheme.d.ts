import 'styled-components';

declare module 'styled-components' {
  export interface DefaultTheme {
    boxShadow: string;
    colors: {
      background: '#FFFFFF';

      cta: string; // buttons, checkboxes, price (i.e. everything screaming on user)
      textPrimary: string;
      textSecondary: string;
    };
  }
}
