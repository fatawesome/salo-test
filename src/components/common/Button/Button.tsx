import React, { MouseEventHandler } from 'react';
import styled from 'styled-components';

interface ButtonProps {
  className?: string;
  onClick: MouseEventHandler;
}

const Button: React.FC<ButtonProps> = (props) => {
  return (
    <button onClick={props.onClick} className={props.className}>
      {props.children}
    </button>
  )
}

const StyledButton = styled(Button)`
  height: 50px;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: ${props => props.theme.colors.cta};
  color: #FFFFFF;
  text-transform: uppercase;
  filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));
  border-radius: 5px;
  border: none;
  
  &:active {
    filter: none;
  }
  &:hover {
    cursor: pointer;
  }
`;

export default StyledButton;
