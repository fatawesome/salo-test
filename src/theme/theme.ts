import { DefaultTheme } from 'styled-components';

// TODO: font and stuff;
const aviasalesTheme: DefaultTheme = {
  boxShadow: '0px 2px 8px rgba(0, 0, 0, 0.1)',
  colors: {
    textPrimary: '#4A4A4A',
    textSecondary: '#A0B0B9',
    background: '#FFFFFF',
    cta: '#2196F3',
  }
}

export { aviasalesTheme };
