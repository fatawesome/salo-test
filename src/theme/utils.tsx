import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { aviasalesTheme } from './theme';
import { render } from '@testing-library/react';

const withTheme = (node: React.ReactNode) => (
  <ThemeProvider theme={aviasalesTheme}>
    {node}
  </ThemeProvider>
)

export const snapshotWithTheme = (node: React.ReactNode) => {
  return renderer.create(withTheme(node));
}

export const uiWithTheme = (node: React.ReactNode) => {
  return render(withTheme(node));
}

// TODO: можно обобщить в функцию вида (fn :: renderer -> node -> result)
