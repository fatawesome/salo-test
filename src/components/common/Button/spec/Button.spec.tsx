import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from 'styled-components';
import { aviasalesTheme } from '../../../../theme/theme';

import Button from '../Button';

describe('Button component', () => {
  const buttonText = 'Click me'
  const clickHandler = jest.fn();

  test('renders correctly', () => {
    const component = renderer.create(
      <ThemeProvider theme={aviasalesTheme}>
        <Button onClick={clickHandler}>{buttonText}</Button>
      </ThemeProvider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  test('is clickable', () => {
    render(
      <ThemeProvider theme={aviasalesTheme}>
        <Button onClick={clickHandler}>{buttonText}</Button>
      </ThemeProvider>
    );
    userEvent.click(screen.getByText(buttonText));
    expect(clickHandler).toBeCalledTimes(1);
  });
})
