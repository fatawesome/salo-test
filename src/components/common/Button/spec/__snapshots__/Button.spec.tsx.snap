// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
<button
  className="Button__StyledButton-sc-1atgevq-0 gBeAQF"
  onClick={[MockFunction]}
>
  Click me
</button>
`;
